### PR TITLE
Add system label overflow

### PR DESCRIPTION
**Pull request by zilongshanren**
for issue: https://github.com/cocos2d/cocos2d-x/issues/15826

Note: currently only iOS, Android and Mac are supported.

This PR mainly add missing `clamp` and `shrink` feature to system label.

### CODE REVIEW

**Comment by minggo** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
iOS platform can't pass CI, I need to add CoreText to other tests project: js-tests/lua-tests etc

**Comment by zilongshanren** [comment]
@dumganhar  please review, thanks.

File: cocos/2d/CCLabel.cpp [hunk 1]
```diff
@@ -560,6 +560,11 @@ void Label::updateShaderProgram()
 
 void Label::setFontAtlas(FontAtlas* atlas,bool distanceFieldEnabled /* = false */, bool useA8Shader /* = false */)
 {
+    if(atlas)
+    {
+        _systemFontDirty = false;
```
Comment (dumganhar) [inline cocos/2d/CCLabel.cpp:565]: Why to set `_systemFontDirty` flag to false here?

File: cocos/base/ccTypes.h [hunk 2]
```diff
@@ -566,7 +568,10 @@ struct CC_DLL FontDefinition
     FontShadow            _shadow;
     /// font stroke
     FontStroke            _stroke;
-
+    /// enable text wrap
+    bool                  _enableWrap;
+    /// enable shrink font size
+    int                  _overflow;
```
Comment (dumganhar) [inline cocos/base/ccTypes.h:577]: Why it's an integer value? If it has to be integer, it's better to add comments to describe the meaning of integer value.

**Comment by zilongshanren** [comment]
@dumganhar  The comments has been improved, please check.

File: cocos/base/ccTypes.h [hunk 3]
```diff
@@ -570,7 +570,10 @@ struct CC_DLL FontDefinition
     FontStroke            _stroke;
     /// enable text wrap
     bool                  _enableWrap;
-    /// enable shrink font size
+    /** There are 4 overflows: none, clamp, shrink and resize_height.
+     *  The conresponding integer values are 0, 1, 2, 3 respectively
+     * For more information, please refer to Label::Overflow enum class.
+     */
     int                  _overflow;
```
Comment (minggo) [inline cocos/base/ccTypes.h:577]: why not use enum instead of int?

Comment (zilongshanren) [inline cocos/base/ccTypes.h:577]: The enum is defined inside Label class, and it need to pass around to JNI. Int is simpler.

**Pull request merged**